x-pack/plugins/maps/public/reducers/map/types.ts

```diff
@@ -35,7 +36,10 @@ export type MapContext = {
   filters: Filter[];
   refreshConfig?: MapRefreshConfig;
   refreshTimerLastTriggeredAt?: string;
+  vectorLayerIndexName: string;
   drawState?: DrawState;
+  shapeToDraw?: DRAW_TYPE;
```

Comment:
see other comment. this feels like we're merging a few concepts. "drawState" (for filter-creation), and edit-state while we're in edit-mode for a layer.

we're modifying `DrawState` to keep track of a few new things, and then add this other state next to it.

I'd suggest we remove the shapeToDraw and featuresToIndexQueue, and collect everything we need for editing in a new state-object `EditState`